workflow/rules/common.smk

```diff
@@ -88,19 +88,29 @@ def get_model(wildcards):
 
 def is_single_end(sample, unit):
     """Determine whether unit is single-end."""
-    bam_paired_not_present = pd.isnull(units.loc[(sample, unit), "bam_paired"])
+    if "bam_paired" in units.columns:
+        bam_paired_not_present = pd.isnull(units.loc[(sample, unit), "bam_paired"])
+    else:
+        bam_paired_not_present = True
     fq2_not_present = pd.isnull(units.loc[(sample, unit), "fq2"])
     return fq2_not_present and bam_paired_not_present
```

Comment:
This might be over-engineering a bit, but we could make the code changes here a little less repetitive by creating the following function:

```
def column_missing_or_empty(column_name, dataframe, sample, unit):
    if column_name in dataframe.columns:
        return pd.isnull(dataframe.loc[(sample, unit), column_name])
    else:
        return True
```

Then the code here becomes:
```suggestion
    return column_missing_or_empty("fq2", units, sample, unit) and column_missing_or_empty("bam_paired", units, sample, unit)
```